New 0.62.0 changelog

kelset opened the pull request:
Allright, this is the new version of https://github.com/react-native-community/releases/pull/156 that uses the new improved script (thanks @alloy!).

As we can clearly see, now the number of changes is quite more massive than the one we saw in the previous one... which means that in the ballpack number of 1.2k commits around 60-ish percent are not FB internals.

But there are still quite a big number of "unknowns" to put in the right place... I don't have time to do it atm but I'm starting to open this as a draft PR.

---
RELEVANT LOGS from the command:
```
Unable to find differential revisions for the following commits. If these were made on the release branch only, be sure to update the CHANGELOG entries to point to the commit on the master branch after back-porting.
    https://github.com/facebook/react-native/commit/be5088401fd8e19d57adda42d275cab437448064
    https://github.com/facebook/react-native/commit/26e7d6a06211412a079af746750c5dd9edbfecbb
    https://github.com/facebook/react-native/commit/cef001713fc6384353bbcb4d45645ceee44ed1a9
    https://github.com/facebook/react-native/commit/99ef72cad73da25d76bf8177272724cfe6cece61
    https://github.com/facebook/react-native/commit/55276a9b104781b49b66938d71aac385bdd236e5
```

and 
```
Unable to find differential revisions for the following commits. If these were made on the release branch only, be sure to update the CHANGELOG entries to point to the commit on the master branch after back-porting.
    https://github.com/facebook/react-native/commit/be5088401fd8e19d57adda42d275cab437448064
    https://github.com/facebook/react-native/commit/26e7d6a06211412a079af746750c5dd9edbfecbb
    https://github.com/facebook/react-native/commit/cef001713fc6384353bbcb4d45645ceee44ed1a9
    https://github.com/facebook/react-native/commit/99ef72cad73da25d76bf8177272724cfe6cece61
    https://github.com/facebook/react-native/commit/55276a9b104781b49b66938d71aac385bdd236e5
```

jeremy-deutsch commented:
I'd like to help sort the "unknowns". @kelset, would it be possible for you to give someone trusted (like @alloy) edit access to this branch/PR, so that you don't have to be the only person who approves suggestions/moves around the "unknowns"?

TheSavior commented:
Can we get the unknown ones grouped by user? Almost all of those are Facebook employees, so we can just ask them to write the changelog for those commits.

TheSavior commented on CHANGELOG.md:
```diff
@@ -1,5 +1,741 @@
+- Remove duplicate flow type for TouchableProps ([fb5276c189](https://github.com/facebook/react-native/commit/fb5276c1897982f0be96f2e7074d839ae8fffdc8) by [@TheSavior](https://github.com/TheSavior))
+- Lists: Fix invalid prop usage in RNTesterExampleList ([da057749f8](https://github.com/facebook/react-native/commit/da057749f8be59516b7dcb5cb3a7bdb67c3d990c) by [@TheSavior](https://github.com/TheSavior))
+- Enable no-useless-escape lint rule ([90977b0e00](https://github.com/facebook/react-native/commit/90977b0e00acc6b3263502017c27094392e89478) by [@motiz88](https://github.com/motiz88))
+- Fix throttling in scroll view ([4159e20146](https://github.com/facebook
```
This seems strange, what happened here?

alloy replied:
Thanks for identifying these! I’ll plug them into the regression test suite and get them sorted 👍

alloy replied:
Fixed in https://github.com/react-native-community/releases/pull/169 and updated this PR.

TheSavior commented on CHANGELOG.md:
```diff
@@ -1,5 +1,741 @@
+- Remove duplicate flow type for TouchableProps ([fb5276c189](https://github.com/facebook/react-native/commit/fb5276c1897982f0be96f2e7074d839ae8fffdc8) by [@TheSavior](https://github.com/TheSavior))
+- Lists: Fix invalid prop usage in RNTesterExampleList ([da057749f8](https://github.com/facebook/react-native/commit/da057749f8be59516b7dcb5cb3a7bdb67c3d990c) by [@TheSavior](https://github.com/TheSavior))
+- Enable no-useless-escape lint rule ([90977b0e00](https://github.com/facebook/react-native/commit/90977b0e00acc6b3263502017c27094392e89478) by [@motiz88](https://github.com/motiz88))
+- Fix throttling in scroll view ([4159e20146](https://github.com/facebook
```
Here too. Same as the below error but also missing an author?

alloy replied:
Fixed in https://github.com/react-native-community/releases/pull/169 and updated this PR.

jeremy-deutsch commented:
It looks like GitHub's having some issues (note the quoted section of the changelog):
![Screenshot from 2020-03-03 11-01-39](https://user-images.githubusercontent.com/13303900/75809930-6c076180-5d3e-11ea-918a-5e190b0fe869.png)

alloy commented:
@TheSavior As a one-off, for just this version, we can sort those sections with `sort -t @ -k 2`. E.g. on macOS, copy the entries of an unknown section and then run:

```bash
pbpaste | sort -t @ -k 2 | pbcopy
```

And paste it back.

youngjuning commented on CHANGELOG.md:
```diff
@@ -1,5 +1,741 @@
+- Removing <TextInput>'s `onTextInput` event ([3f7e0a2c96](https://github.com/facebook/react-native/commit/3f7e0a2c9601fc186f25bfd794cd0008ac3983ab) by [@shergin](https://github.com/shergin))
+- Remove TextInput's `inputView` prop ([1804e7cbea](https://github.com/facebook/react-native/commit/1804e7cbea707a35046118090966a54031edfae8) by [@TheSavior](https://github.com/TheSavior))
+- Animated: Remove `defaultProps` Parameter ([a70987cee2](https://github.com/facebook/react-native/commit/a70987cee24bcd027b9c4a5aa85dfd6a1aab74b3) by [@yungsters](https://github.com/yungsters))
+- RN: Delete `TextInput.selectionState` Prop ([2becdfd404](https://github.com/facebook/react-native/commit/2becdfd4041f7f28138ba3a61c03e17c06dc2e50) by [@yungsters](https://github.com/yungsters))
```
How can i fit this change?

alloy replied:
@youngjuning I’m not sure what the question is. If you are asking about the linked change, then I would suggest asking the question on that commit instead; this PR is purely about the CHANGELOG.

alloy commented:
Btw, we should probably only fix the unknown entries once we generate the changelog for the version we agree to release as stable version, right? Then people can just PR their fixed changelog entries onto this branch.

TheSavior commented on CHANGELOG.md:
```diff
@@ -1,5 +1,741 @@
+- Remove duplicate flow type for TouchableProps ([fb5276c189](https://github.com/facebook/react-native/commit/fb5276c1897982f0be96f2e7074d839ae8fffdc8) by [@TheSavior](https://github.com/TheSavior))
+- Lists: Fix invalid prop usage in RNTesterExampleList ([da057749f8](https://github.com/facebook/react-native/commit/da057749f8be59516b7dcb5cb3a7bdb67c3d990c) by [@TheSavior](https://github.com/TheSavior))
+- Enable no-useless-escape lint rule ([90977b0e00](https://github.com/facebook/react-native/commit/90977b0e00acc6b3263502017c27094392e89478) by [@motiz88](https://github.com/motiz88))
+- Fix throttling in scroll view ([4159e20146](https://github.com/facebook
```
This one shows up as a syntax error in Sublime because of the use of brackets in the commit message implying the beginning of a link. Not sure if it is actually a problem or not but I wanted to flag.

alloy replied:
I suspect Sublime is trying to be helpful about there being a space between `[2/N]` and the subsequent open parenthesis? It renders as we want it to, at least on GitHub.

kelset commented:
re: @jeremy-deutsch 
> I'd like to help sort the "unknowns". @kelset, would it be possible for you to give someone trusted (like @alloy) edit access to this branch/PR, so that you don't have to be the only person who approves suggestions/moves around the "unknowns"?

Eloy has write access to this repo so he can "take over" this branch when/if he wants :)
As do all the members of the RNComm github org, since it's the default setting to have write access to all repos.

I don't think I will have time to work on the feedback here today - so if anyone is taking it over please just write a comment when they do and when they are done so that we are all aware 🤗

alloy commented:
@TheSavior Updated based on RC4 and sorted the unknowns by author.

TheSavior commented:
Yay! Thanks

Do you think this is likely to be the one and we should get our team to start annotating the uncategorized commits? Or should we hold off?

alloy commented:
I’d like for this to be the one, so yeah ready to get started 👍

rickhanlonii commented:
@kelset do you want to review the suggested changes and I'll take a pass after that?

TheSavior commented:
@rickhanlonii, I think there was talk of someone just making changes and committing back to this branch. Do you have access?

I imagine that is easier than flagging 100 commits as internal for someone else to remove :-)

alloy commented:
@jeremy-deutsch Hiya! You seem interested in helping out with the CHANGELOG, which is absolutely welcome 👌 

I don’t believe anybody has started on this _yet_ (correct me if I’m wrong @TheSavior), so please feel free to make a PR into this branch. It would probably be best for now to limit changes to moving around those that are in the unknown/wrong section and leave the internal ones to FB people to resolve.

As for the folding together of details about new features such as Flipper; I generally prefer that, but will defer to the people that have been doing these releases longer than I have.

jeremy-deutsch commented:
@alloy Sounds good to me! I’ll take a look to see if anything might be better categorized, but hold off on marking anything else as internal or non-stable-RN. The big bucket of suggestion comments might not have been the perfect approach here 🙂

alloy commented:
Ace, thanks 🙏

kelset commented:
hey folks - I should have some time today so I'll tackle this in the next couple hours. @jeremy-deutsch I really want to thank you for all your feedback! And don't worry, leaving all those comments is ok - luckily now GH has some nice UX over hiding addressed comments.

I'll try to merge everything you proposed, then do another cleanup, then report back - so that you don't have to waste time re-addressing all the changes you have proposed as a PR 👍

kelset commented on CHANGELOG.md:
```diff
@@ -1,5 +1,699 @@
+- RCTRedBox ability to add extra buttons ([fc80a51457](https://github.com/facebook/react-native/commit/fc80a5145719ccecbefc25bfedbad6224b54efab) by [@mmmulani](https://github.com/mmmulani))
+- Make RCTAsyncLocalStorage TurboModule-compatible ([2b62bd7695](https://github.com/facebook/react-native/commit/2b62bd7695e568ff97797f161ef9f75a063eece2) by [@RSNara](https://github.com/RSNara))
+- Make RCTAlertManager TurboModule-compatible ([da6274ba76](https://github.com/facebook/react-native/commit/da6274ba76afaa271d6a74e4143479c93fa9bb14) by [@RSNara](https://github.com/RSNara))
+- Add definition for `queryCache` in ImageLoader ([ea1e8bbd34](https://github.com/facebook/react-native/commit/ea1e8bbd34b655a062ef13b6c2c76660ed43d31a) by [@lunaleaps](https://github.com/lunaleaps))
```
TBH I'm aware of some people actively using the JSI directly, so I'd prefer to keep this in.

kelset commented:
Hey folks. I've taken a stab at improving the status of the PR and I'm done for the day; here's the breakdown:
* I've merged most suggestions from @jeremy-deutsch 
* rebased this PR on top of latest master
* addressed a couple of typos and other minor things
* added a draft for the intro section

That said, for this PR to be ready there is one major thing that still needs to be done: meaning, addressing the huge unknown section. I'm not sure who is the best person for that at this moment but we'll discuss it asap.

Aside from that, to address the only other comment left:
>  I realize the format of the changelog is to list by commit, but maybe all the Flipper-related commits should be one changelog item.

I personally prefer to have multiple commits regarding Flipper in here - having just one line for all the work being done around it may confuse some people around some details of how it has been addded/implemented. But I'm more than happy to consider alternative options such as creating a one-off dedicated section with all the Flipper related commits.

TheSavior commented:
I believe @rickhanlonii said he would take point on addressing the unknown commits from the Facebook side

alloy commented on CHANGELOG.md:
```diff
@@ -2,6 +2,19 @@
 
+This feature release of React Native includes many changes for the platform: it has been a lot of work, so please refer to the [blog post](ADD) for overlook of the main changes - like Flipper support and the new LogBox.
+
+We also know that these past few weeks have been incredibly challenging for all of us, all over the world. The impact of COVID-19 is still reshaping our lives as we write these lines, and we hope that you will not feel pressured into upgrading to this latest. We went through more testing than ever to make sure that this new 62 will be a pleasant experience for everyone, but we are not perfect; so ...
```
I’d be in favour of a little less appeasing and being a bit more matter of fact. For instance, something more to the point like:

```suggestion
As always, we’ve done our best to test this release at length, especially with this being such a large release. However, note that due to challenges caused by COVID-19 to people everywhere, we [the maintainers] may not have the bandwidth we’d like to have for quick follow-up on bugs and/or a patch release. When you run into an issue, please do try to provide a fix or a detailed ticket, but be prepared to revert to the version you were previously on to unblock yourself.
```

kelset replied:
cool - adding it.

jeremy-deutsch commented:
> I personally prefer to have multiple commits regarding Flipper in here - having just one line for all the work being done around it may confuse some people around some details of how it has been addded/implemented. But I'm more than happy to consider alternative options such as creating a one-off dedicated section with all the Flipper related commits.

Yeah, with "one changelog item" I was thinking something more along the lines of
```
- Add Flipper support ([abcdef] by @this-person, [123456] by @that-person, ...)
```
so we still have all the commits for reference, but they're not spread around the changelog. There's a couple other examples of this in the changelog already, for example in v0.60->Added->New Intro screen.

rickhanlonii commented:
Yeah I'm going to go over this and all the unknowns, was just waiting for @jeremy-deutsch's changes to be applied. Last night I [improved the script](https://github.com/react-native-community/releases/pull/173) to find commits that mention a changelog but were formatted incorrectly which will make it easier to go through the unknowns.

minderov commented on CHANGELOG.md:
```diff
@@ -1,9 +1,675 @@
+- Slightly more aggressive FabricUIManager teardown ([fa1c2be387](https://github.com/facebook/react-native/commit/fa1c2be387a95d90d8fcdaabb88b1c4c61e0097b) by [@JoshuaGross](https://github.com/JoshuaGross))
+- Support for UpdatePaddingMountItem ([44bfc4be44](https://github.com/facebook/react-native/commit/44bfc4be447680096f01fd00a1344c2cd398a981) by [@JoshuaGross](https://github.com/JoshuaGross))
+- Unbreak master ([a1ee1d7698](https://github.com/facebook/react-native/commit/a1ee1d7698081e7882d2fc5f08d848f93623f1c4) by [@JoshuaGross](https://github.com/JoshuaGross))
+- Work around crashing redbox ([43f042eb7d](https://github.com/facebook/react-native/commit/43f042eb7db0f
```
Should this change be under the `#### iOS Unknown` category?

jeremy-deutsch replied:
The `Unknown` category isn’t actually part of the normal changelog - it’s items that haven’t been categorized yet. Ideally, all the `Unknown` items should all be gone when the changelog is finalized, so we’re not moving anything else into there

alloy commented:
@jeremy-deutsch To follow up on my earlier comment, if you’d like to be more actively involved in work like this, please reach out to me at `eloy.de.enige@gmail.com`

rickhanlonii commented:
Ok all the unknowns have been categorized or removed 👍

kelset commented on CHANGELOG.md:
```diff
@@ -1,9 +1,234 @@
+
+This feature release of React Native includes many changes for the platform: it has been a lot of work, so please refer to the [blog post](ADD) for overlook of the main changes - like Flipper support and the new LogBox.
+
+As always, we’ve done our best to test this release at length, especially with this being such a large release. However, note that due to challenges caused by COVID-19 to people everywhere, we [the maintainers] may not have the bandwidth we’d like to have for quick follow-up on bugs and/or a patch release. When you run into an issue, please do try to provide a fix or a detailed ticket, but be pre ...
```
@alloy just a super quick question, in this phrase 
> [the maintainers]

is this supposed to be a link to something or it was more like " we = the maintainers"?

alloy replied:
The latter 👍

kelset commented:
Thanks to everyone for their comments, it looks like we are (almost!) ready to merge this.

Only two things missing are the links to the blogpost (still not up) and to the dedicated issue (still not open).

Anything else that I may be forgetting?

jeremy-deutsch commented:
@kelset I have some nits (mostly consistency, plus a couple items were accidentally duplicated), but nothing blocking

kelset commented:
thanks @jeremy-deutsch! Merged 👍

alloy commented on CHANGELOG.md:
```diff
@@ -1,9 +1,223 @@
+If you're upgrading, manual intervention may be required for your app. Please see the [upgrade-helper](https://react-native-community.github.io/upgrade-helper/) for a detailed breakdown of the changes required and see [this issue](https://github.com/react-native-community/releases/issues/179) for known issues.
+
+Note two known issues with their corresponding workarounds:	
+- [Release version instacrash on Android](https://github.com/facebook/react-native/issues/28163)	
```
Btw, 👴 here, is ‘instacrash’ a word that helps people understand this sentence?

kelset replied:
I just copied the title of the issue, if there's a better way of phrasing it we could change it here and in the issue.

Maybe something like `Crash on Android when creating a release version`?

alloy replied:
What about?

```suggestion
- [Crash on launch of Android release builds](https://github.com/facebook/react-native/issues/28163)	
```

alloy replied:
Hmm, in retrospect, I thought this was about another issue–namely https://github.com/facebook/react-native/issues/28270.

I actually don’t think we should link to https://github.com/facebook/react-native/issues/28163, as it does not appear to be a problem affecting multiple people. All commenters seem to have different crash logs and the original poster seems to be using Vue (?).

alloy replied:
Alright, removed this 👍

HelloCore commented on CHANGELOG.md:
```diff
@@ -1,9 +1,223 @@
+- Fix ReactInstanceManagerBuilder.build crashing if SoLoader has not been explicitly initialized ([60e00d9d96](https://github.com/facebook/react-native/commit/60e00d9d96d7b186c1d4c1542caddc1b74eeb3da) by [@petterh](https://github.com/petterh))
+- Fix default accessibility hint not being read. ([f8dff0bcb3](https://github.com/facebook/react-native/commit/f8dff0bcb3147b7a1aa8ac7159952d848e198e29))
+- Fix JS bundle loading progress bar ([7b9d6d19e2](https://github.com/facebook/react-native/commit/7b9d6d19e2c0854aa53587ef68ce715fb7803e2a) by [@makovkastar](https://github.com/makovkastar))
+- AFix ndroid Q related NaN error - don't try to do math with NaN values ([db5994980d](https://github.com/facebook/react-native/commit/db5994980df136c5cce6cd90348b4bf18180562f))
```
A small typo here.
```suggestion
- Fix Android Q related NaN error - don't try to do math with NaN values ([db5994980d](https://github.com/facebook/react-native/commit/db5994980df136c5cce6cd90348b4bf18180562f))
```

alloy replied:
Thanks!

thymikee commented on CHANGELOG.md:
```diff
@@ -1,9 +1,221 @@
 
+## v0.62.0
+
+This major release includes Flipper support by default, improved dark mode support, moving Apple TV to [react-native-tvos](https://github.com/react-native-community/react-native-tvos), and more. See the [blog post](https://reactnative.dev/blog/2020/03/24/version-0.62) for all of the highlights.
```
```suggestion
This major release includes Flipper support by default, improved dark mode support, moving Apple TV to [react-native-tvos](https://github.com/react-native-community/react-native-tvos), and more. See the [blog post](https://reactnative.dev/blog/2020/03/26/version-0.62) for all of the highlights.
```

thymikee commented on CHANGELOG.md:
```diff
@@ -1,9 +1,221 @@
+- Limit size of video uploaded from camera roll in android (< 100 MB) ([d21f695edf](https://github.com/facebook/react-native/commit/d21f695edf367166a03af4c6e9376cd498b38665))
+- Remove "Reload on JS change" from RN Android ([478df155e7](https://github.com/facebook/react-native/commit/478df155e70a4ce30219adcac6f0801c4e4d10ec) by [@cpojer](https://github.com/cpojer))
+
+#### iOS specific
```
this is empty, so remove?
